packages/ODEs/src/app.ts

```diff
@@ -1,654 +1,870 @@
-// Application for solving initial value problems (IVP) & demo app
+// UI for solving differential equations
 
 import * as grok from 'datagrok-api/grok';
 import * as ui from 'datagrok-api/ui';
 import * as DG from 'datagrok-api/dg';
 
-import {basicSetup, EditorView} from "codemirror";
-import {EditorState} from "@codemirror/state";
-import {python} from "@codemirror/lang-python";
-import {autocompletion} from "@codemirror/autocomplete";
+import {basicSetup, EditorView} from 'codemirror';
+import {EditorState} from '@codemirror/state';
+import {python} from '@codemirror/lang-python';
+import {autocompletion} from '@codemirror/autocomplete';
 
 import {DF_NAME, CONTROL_EXPR, MAX_LINE_CHART} from './constants';
 import {TEMPLATES, DEMO_TEMPLATE} from './templates';
-import { USE_CASES } from './use-cases';
-import {HINT, TITLE, LINK, HOT_KEY, ERROR_MSG, INFO, WARNING, MISC, demoInfo as demoAppInfo} from './ui-constants';
-import {getIVP, getScriptLines, getScriptParams} from './scripting-tools';
+import {USE_CASES} from './use-cases';
+import {HINT, TITLE, LINK, HOT_KEY, ERROR_MSG, INFO,
+  WARNING, MISC, demoInfo, INPUT_TYPE, PATH, TIMEOUT} from './ui-constants';
+import {getIVP, getScriptLines, getScriptParams, IVP, Input, SCRIPTING,
+  BRACE_OPEN, BRACE_CLOSE, BRACKET_OPEN, BRACKET_CLOSE, ANNOT_SEPAR,
+  CONTROL_SEP, STAGE_COL_NAME, ARG_INPUT_KEYS} from './scripting-tools';
 
 /** State of IVP code editor */
 enum EDITOR_STATE {
-  CLEAR = 0,
-  BASIC_TEMPLATE = 1,
-  ADVANCED_TEMPLATE = 2,  
-  FROM_FILE = 3,
-  EXTENDED_TEMPLATE = 4,
-  CHEM_REACT = 5,
-  ROBERT = 6,
-  FERM = 7,
-  PKPD = 8,
-  ACID_PROD = 9,
-  NIMOTUZUMAB = 10,
+  EMPTY = 'empty',
+  BASIC_TEMPLATE = 'template',
+  ADVANCED_TEMPLATE = 'advanced',
+  FROM_FILE = 'from-file',
+  EXTENDED_TEMPLATE = 'extended',
+  CHEM_REACT = 'chem-react',
+  ROBERT = 'robertson',
+  FERM = 'fermentation',
+  PKPD = 'pk-pd',
+  ACID_PROD = 'ga-production',
+  NIMOTUZUMAB = 'nimotuzumab',
 };
 
-/** Get problem with respect to IVP editor state. */
+/** Models & templates */
+const MODELS: string[] = [EDITOR_STATE.BASIC_TEMPLATE,
+  EDITOR_STATE.ADVANCED_TEMPLATE,
+  EDITOR_STATE.EXTENDED_TEMPLATE,
+  EDITOR_STATE.CHEM_REACT,
+  EDITOR_STATE.ROBERT,
+  EDITOR_STATE.FERM,
+  EDITOR_STATE.PKPD,
+  EDITOR_STATE.ACID_PROD,
+  EDITOR_STATE.NIMOTUZUMAB,
+];
+
+/** Get problem with respect to IVP editor state */
 function getProblem(state: EDITOR_STATE): string {
   switch (state) {
-    case EDITOR_STATE.BASIC_TEMPLATE:
-      return TEMPLATES.BASIC;
+  case EDITOR_STATE.BASIC_TEMPLATE:
```

Comment:
maybe it's a minor detail. but switches almost always can be reduced
you can make an object like `{[key: string]: string}` (you can look at that detail in platform-viewers-demo.ts in the Tutorials package), and then use `arrayName[state]` to get the value
so you will have less code and less logic in this situation

Reply:
![image](https://github.com/datagrok-ai/public/assets/61393026/777a6676-00cc-4575-8952-acbd13517fe1)

Reply:
Replace by the map object